File: TODOList.Domain/Models/RequestModels/CommandRequestModels/UpdateTaskItemCommandRequestModel.cs
using MediatR;

namespace TODOList.Domain.Models.RequestModels.CommandRequestModels
{
    public class UpdateTaskItemCommandRequestModel : IRequest
    {
        public string Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public bool IsDone { get; set; }
    }
}

File: frontend/src/app/models/request/command/updateTaskItemCommandRequestModel.ts
export default class UpdateTaskItemCommandRequestModel {
    constructor(id: string, name: string, isDone: boolean, description?: string) {
        this.id = id
        this.name = name
        this.isDone = isDone
        this.description = description
    }

    id: string
    name: string
    description?: string
    isDone: boolean
}
